synapse/lib/view.py

```diff
@@ -275,6 +275,27 @@ def init2(self):
     def isafork(self):
         return self.parent is not None
 
+    def isForkOf(self, viewiden):
+        view = self.parent
+        while view is not None:
```

Comment:
we'd just need to maintain a set of view's we've visited and if we see that iden a second time we know something is very broken. constructing a circular view dependency would, I believe, require hand editing the data structures in the hive, which is "here be dragons" territory.

Reply:
If you were to manage to construct a circular parent list, this would be the least of your troubles :D  Any storm query issued against that view would wedge the cortex forever :)